### ISSUE TITLE
Change the default location of the exchange directory

### ISSUE THREAD

**Issue by jhamrick** [jupyter/nbgrader#841]
Currently, it is at `/srv/nbgrader/exchange`, but this is likely a location that is owned by root and cannot be created by the user. A better default location would be in `/usr/local/share/nbgrader/exchange`, but this might break existing installations that assume the exchange is in `/srv`. It might not be too bad to detect if the exchange isn't being set in the config, and warn users that the location has moved.

Thoughts @ellisonbg @lgpage ?

**Comment by lgpage** [comment]
From the FHS (http://www.pathname.com/fhs/):

> `/srv` contains site-specific data which is served by this system. This main purpose of specifying this is so that users may find the location of the data files for particular service, and so that services which require a single tree for readonly data, writable data and scripts (such as cgi scripts) can be reasonably placed. Data that is only of interest to a specific user should go in that users’ home directory.

> `/usr/local/share` - The requirements for the contents of this directory are the same as `/usr/share`. The `/usr/share` hierarchy is for all read-only architecture independent data files. Any program or package which contains or requires data that doesn’t need to be modified should store that data in `/usr/share` (or `/usr/local/share`, if installed locally). It is recommended that a subdirectory be used in `/usr/share` for this purpose.

My recommendation is to leave it as is, but a nbgrader only usage (plus the webapp) should work without needing to set, read, or write to `/srv` and only the exchange specific commands (release, collect, etc.) should fail indicating `/srv/nbgrader/exchange` does not exist / un-writable (I.e. nbgrader has not been set up correctly for server usage). The "Assignments" tab in the Jupyter Dashboard (from the Assignment List extension) should probably also give a similar indication.

Any server and or JupyterHub set-up that runs nbgrader I would assume is run by root?

Related: #820

**Comment by jhamrick** [comment]
Yeah, based on that spec it does sound like `/srv` is more appropriate. I guess it's a bit frustrating for users because they need root to create the directory (whereas they wouldn't in `/usr/local/share`). It also seems odd to me to require root for a default configuration setting.

Everything about nbgrader *should* work ok now if the exchange isn't set up, rather than breaking, and will report via a log message that the exchange doesn't exist.

> Any server and or JupyterHub set-up that runs nbgrader I would assume is run by root?

JupyterHub does need to be setup as root, but a particular instructor on a JupyterHub instance might not necessarily have root access. So that means they'd have to ask a sysadmin to create the exchange for them, or set it to a different location in the config. Either way, it involves an additional step that they have to figure out.

**Comment by dkirkby** [comment]
A new install currently displays this prominent warning on the Formgrader page, which is a bit disconcerting:
```
The exchange directory at /srv/nbgrader/exchange does not exist and could not be created.
The "release" and "collect" functionality will not be available. Please see the documentation
on Setting Up The Exchange for instructions.
```
Is there a way to silence this if you don't need "release" and "collect", e.g., by setting:
```
c.Exchange.root = None
```
in `nbgrader_config.py`?

**Comment by jhamrick** [comment]
I think that's a reasonable request and wouldn't be too hard to implement. I'll open a separate issue for it to track it.

**Comment by ellisonbg** [comment]
It sounds like `/srv` is the right location. In the cases where an exchange directory is being used, I do think it is reasonable to expect that someone with root access is setting things up. I do think we should make it possible to run nbgrader with no exchange and no warnings like this and something like the above recommendation seems like a good approach.

**Comment by jhamrick** [comment]
Ok, given the discussion here, I am going to go ahead and close this issue since it sounds like there's consensus we shouldn't change the default location.

**Comment by nscozzaro** [comment]
Hi @jhamrick, I recently discovered NbGrader and it seems really cool! To test it out I just set up a JupyterHub and am using the `umsimads/education-notebook/` docker image which comes with nbgrader installed. However, I received the same error as discussed above (`The exchange directory at /srv/nbgrader/exchange does not exist...`). I read the page the error message links to (https://nbgrader.readthedocs.io/en/stable/user_guide/managing_assignment_files.html#setting-up-the-exchange), and executed the commands on that page and then refreshed my nbgrader page but I still see the same message. I also read through the JupyterHub config page (https://nbgrader.readthedocs.io/en/stable/configuration/jupyterhub_config.html) but unfortunately I'm still confused about what to do! 

What exactly do I need to configure in order to get it the exchange to work on my JupyterHub? Thanks in advance!


![Screen Shot 2019-08-19 at 10 13 49 PM](https://user-images.githubusercontent.com/15185105/63312186-ad6a7f00-c2ce-11e9-982c-a47ee3cc08cb.png)
